### ISSUE TITLE
Question: save the compile in PKG_CACHE_PATH and use it on next try

### ISSUE THREAD

**Issue by biancode** [zeit/pkg-fetch#87]
In my case on ARM (PI3) it builds every time node again and again from Git. There is a built file named built-v12.16.1-linux-armv7, but next pkg call compiles again. Is there some I can do to step over the compile after it is done once?

**Comment by biancode** [comment]
Parameter using:

```
        forceFetch
        forceBuild
        dryRun
```

**Comment by biancode** [comment]
pkg goes to search with `dryRun: true` and get with this every time ?nothing? - dead look situation

```
            if (!dryRun) {
              _context.next = 34;
              break;
            }
```

**Comment by igorklopov** [comment]
I failed to reproduce. Both with custom PKG_CACHE_PATH and without it - `built` base binary is used, without fetching and without downloading. RPi3. pkg@4.4.6. Please can you elaborate detailed steps to reproduce the issue.

**Comment by biancode** [comment]
I tested more and it was a volume problem with docker. Issue was me.

**Comment by biancode** [comment]
AssertionError [ERR_ASSERTION]: Placeholder for not found - is mostly a .pkg-cache folder problem

fixed via: ` export PKG_CACHE_PATH=your-pwd/.pkg-cache` 
check with: `echo $PKG_CACHE_PATH`
and do `cd $PKG_CACHE_PATH && chmod  755 *` execute to set execute rights to all ( a+x )

**Comment by biancode** [comment]
Placeholder for not found means notFound variable is true - maybe the message could tell more here to know why it is failing. Some like: "Some was not found. Please, check A, B and maybe C" :)

**Comment by parkan** [comment]
@biancode could you explain your fix a bit more? I am seeing the same issue in our gitlab CI/CD runner but not on my home machine

relevant dockerfile is like this:

```Docker
FROM node:16-alpine AS node-build
WORKDIR /build

ENV PKG_CACHE_PATH /build/.pkg-cache
RUN mkdir -p ${PKG_CACHE_PATH}
RUN echo ${PKG_CACHE_PATH}
RUN npm i -g pkg ipfs-car
RUN pkg --debug --targets=node16-linuxstatic-x64 `which ipfs-car`
```

**Comment by biancode** [comment]
@parkan it was a problem of missing env variables and in some cases the env variables need to be send to the next shell level or calling shell env, if a new shell starts inside a process. I did not research more, but sometimes it looked like the env variable is empty. Try to make console outputs, while you use needed env variables, which value they have in that state. That was the way, how I found the problem and to solved it.
